src/Persistence_matrix/include/gudhi/matrix.h

```diff
@@ -0,0 +1,2068 @@
+   *
+   * @return A reference to the barcode. The barcode is a vector of @ref Matrix::Bar. A bar stores three informations:
+   * the @ref PosIdx birth index, the @ref PosIdx death index and the dimension of the bar.
+   */
+  const barcode_type& get_current_barcode();
+  /**
+   * @brief Returns the current barcode of the matrix. Available only if
+   * @ref PersistenceMatrixOptions::has_column_pairings is true.
+   *
+   * Recall that we assume that the boundaries were inserted in the order of filtration for the barcode to be valid.
+   *
+   * @warning For simple @ref boundarymatrix "boundary matrices" (only storing \f$ R \f$), we assume that
+   * @ref get_current_barcode is only called once the matrix is completed and won't be modified again.
+   *
+   * @return A const reference to the barcode. The barcode is a vector of @ref Matrix::Bar. A bar stores three
+   * informations: the @ref PosIdx birth index, the @ref PosIdx death index and the dimension of the bar.
+   */
+  const barcode_type& get_current_barcode() const;
+
+  /**
+   * @brief Only available for @ref basematrix "base matrices" without column compression and simple
+   * @ref boundarymatrix "boundary matrices" (only storing \f$ R \f$) and if
+   * @ref PersistenceMatrixOptions::has_column_and_row_swaps is true. Swaps the two given columns. Note for
+   * @ref boundarymatrix "boundary matrices", that it really just swaps two columns and does not update anything else,
+   * nor performs additions to maintain some properties on the matrix.
+   *
+   * @param columnIndex1 First column @ref MatIdx index to swap.
+   * @param columnIndex2 Second column @ref MatIdx index to swap.
+   */
+  void swap_columns(index columnIndex1, index columnIndex2);
+  /**
+   * @brief Only available for @ref basematrix "base matrices" without column compression and simple
+   * @ref boundarymatrix "boundary matrices" (only storing \f$ R \f$) and if
+   * @ref PersistenceMatrixOptions::has_column_and_row_swaps is true. Swaps the two given rows. Note for
+   * @ref boundarymatrix "boundary matrices", that it really just swaps two rows and does not update anything else,
+   * nor performs additions to maintain some properties on the matrix.
+   *
+   * @param rowIndex1 First @ref rowindex "row index" to swap.
+   * @param rowIndex2 Second @ref rowindex "row index" to swap.
+   */
+  void swap_rows(index rowIndex1, index rowIndex2);
+  //TODO: find better name. And benchmark also to verify if it is really worth it to have this extra version in addition
+  //to vine_swap.
+  /**
+   * @brief Only available if @ref PersistenceMatrixOptions::has_vine_update is true and if it is either a bounary
+   * matrix or @ref PersistenceMatrixOptions::column_indexation_type is set to @ref Column_indexation_types::POSITION.
+   * Does the same than @ref vine_swap, but assumes that the swap is non trivial and therefore skips a part of the case
```

Comment:
Is it obvious what "non-trivial" means?

Reply:
Here "non-trivial" means that the we cannot just swap the two columns, i.e. modifications in the matrix have to be done for it to remain valid. If the user has a bit of information about the columns, he can avoid doing quite a few swaps under the right condition (like we do for zigzag) and then this method comes in handy. 
I find "non-trivial" clear, but I am perhaps a bit biased. But for sure I don't like the name of the method, I want to call it differently. I just don't have a good idea yet.